File: C/weightedaverage.c
#include <stdio.h>

int main() {
    double grade1, grade2, grade3;
    double weight1, weight2, weight3;
    double weightedAverage;

    printf("Enter grade 1: ");
    scanf("%lf", &grade1);
    printf("Enter weight 1: ");
    scanf("%lf", &weight1);

    printf("Enter grade 2: ");
    scanf("%lf", &grade2);
    printf("Enter weight 2: ");
    scanf("%lf", &weight2);

    printf("Enter grade 3: ");
    scanf("%lf", &grade3);
    printf("Enter weight 3: ");
    scanf("%lf", &weight3);

    weightedAverage = (grade1 * weight1 + grade2 * weight2 + grade3 * weight3) / (weight1 + weight2 + weight3);

    printf("Weighted average: %.2lf\n", weightedAverage);

    return 0;
}

File: Java/WeightedAverage.java
import java.util.Scanner;

public class WeightedAverage{
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter three grades separated by spaces: ");
        double grade1 = scanner.nextDouble();
        double grade2 = scanner.nextDouble();
        double grade3 = scanner.nextDouble();

        System.out.print("Enter three weights separated by spaces: ");
        double weight1 = scanner.nextDouble();
        double weight2 = scanner.nextDouble();
        double weight3 = scanner.nextDouble();

        double weightedAverage = calculateWeightedAverage(grade1, weight1, grade2, weight2, grade3, weight3);

        System.out.println("The weighted average is: " + weightedAverage);

        scanner.close();
    }

    private static double calculateWeightedAverage(double grade1, double weight1, double grade2, double weight2, double grade3, double weight3) {
        double sumWeights = weight1 + weight2 + weight3;
        double weightedAverage = (grade1 * weight1 + grade2 * weight2 + grade3 * weight3) / sumWeights;
        return weightedAverage;
    }
}
